cmd/appengine/device.go

```diff
@@ -299,14 +360,46 @@ func devicesGetSamplesF(command *cobra.Command, args []string) error {
 			os.Exit(1)
 		}
 
-		for _, v := range page {
-			fmt.Println(prettyPrintDatastreamValue(v))
-			printedValues++
-			if printedValues >= limit && limit > 0 {
-				return nil
+		if outputType == "json" {
+			jsonOutput = append(jsonOutput, page...)
+		} else {
+			for _, v := range page {
+				t.AppendRow([]interface{}{timestampForOutput(v.Timestamp, outputType), v.Value})
+				printedValues++
+				if printedValues >= limit && limit > 0 {
+					renderTableForOutput(t, outputType)
+					return nil
+				}
 			}
 		}
 	}
+	renderTableForOutput(t, outputType)
```

Comment:
What about
```
renderOutput(t, jsonOutput, outputType)
```
so you can hide the `json` output stuff there

Reply:
:+1: with Riccardo comment